lib/web_ui/test/engine/platform_dispatcher/platform_dispatcher_test.dart

```diff
@@ -37,6 +37,21 @@ void testMain() {
       engineDispatcher.dispose();
     });
 
+    test('app lifecycle state has the correct value', () {
+      final MockAppLifecycleStateSupport mockAppLifecycleState =
+          MockAppLifecycleStateSupport();
+      AppLifecycleStateSupport.instance = mockAppLifecycleState;
+      final EnginePlatformDispatcher engineDispatcher =
+          EnginePlatformDispatcher();
+
+      expect(engineDispatcher.appLifecycleState, ui.AppLifecycleState.resumed);
+      mockAppLifecycleState.state = ui.AppLifecycleState.paused;
+      mockAppLifecycleState.invokeListeners(mockAppLifecycleState.state);
+      expect(engineDispatcher.appLifecycleState, ui.AppLifecycleState.paused);
```

Comment:
It'd be nice if this tested that the `state` is propagated through the `flutter/lifecycle` method channel. Also that if the `state` is the same twice, the second time the call is skipped.

Reply:
(If the class gets reimplemented as above, one could just `verify` that the function that is being passed to the created instance is being called with the correct `state` value.)

Reply:
> It'd be nice if this tested that the `state` is propagated through the `flutter/lifecycle` method channel. Also that if the `state` is the same twice, the second time the call is skipped.

Can you suggest how to test this?